fathom/src/surface/elaboration/unification.rs

```diff
@@ -301,17 +311,19 @@ impl<'arena, 'env> Context<'arena, 'env> {
         &mut self,
         spine0: &[Elim<'arena>],
         spine1: &[Elim<'arena>],
-    ) -> Result<(), Error> {
+    ) -> Result<(), Error<'arena>> {
         if spine0.len() != spine1.len() {
-            return Err(Error::Mismatch);
+            return Err(Error::SpineMismatch);
         }
         for (elim0, elim1) in Iterator::zip(spine0.iter(), spine1.iter()) {
             match (elim0, elim1) {
                 (Elim::Fun(input_expr0), Elim::Fun(input_expr1)) => {
                     self.unify(input_expr0, input_expr1)?;
                 }
                 (Elim::Record(label0), Elim::Record(label1)) if label0 == label1 => {}
-                (_, _) => return Err(Error::Mismatch),
+                (_, _) => {
+                    return Err(Error::SpineMismatch);
+                }
```

Comment:
This could be an `Error::ElimMismatch(elim0, elim1)`, perhaps if we were planning on preserving more information in the unification errors.

Reply:
How would we go about using `elim0, elim1` to provide more information? Part of the reason I was exploring tests for test cases was to be able to see the errors produced in these cases, but as you know that proved difficult.